typing/ctype.ml

```diff
@@ -2271,7 +2271,7 @@ let type_jkind_purely_if_principal' =
    don't. *)
 let rec estimate_type_jkind ~expand_component env ty =
   match get_desc ty with
-  | Tvar { jkind } -> Jkind.disallow_right jkind
+  | Tvar { jkind } | Tof_kind jkind -> Jkind.disallow_right jkind
```

Comment:
This is a bit too quiet for my tastes. Just make a new case, please, like `Tunivar`.